### ISSUE TITLE
Low GPU utilization with PyTorch

### ISSUE THREAD

**Issue by darkcristle** [xinntao/Real-ESRGAN#228]
Hello all,

I recently switched from the NCNN implementation to the Pytorch version after encountering a weird tile glitch that I was unable to solve.  Thankfully the same error does not appear to occur on the Pytorch version so that's a plus.  The main "issue" is the speed of the upscale.  It is drastically slower than the Vulkan version with identical arguments.  

I have noticed a disparity in GPU usage between the two versions.  Vulkan seems to use 20-30% whereas the PyTorch version seems to cap at between 10-15%.  Is this simply how it works or have I missed a step where I was supposed to allocate resources?

I had a hypothesis for this but don't know enough to be able to confirm if it is the case.

When I first began using the PyTorch version I discovered that FFmpeg has extracted all frames as 16bit pngs for some reason.  I changed this so that they would now be extracted as 8bit jpg (the original source file is 8bit), this sped things up significantly however it also seems to have caused the GPU issue.  When it was processing 16bit png's utilization peaked at 90%.   Is there any reason that a switch in file format would cause this?

I did some googling and it seems that CUDA prefers raw data in either .raw or .ppm.  Unfortunately, it does not appear that either of these are accepted as valid by the implementation.

Thanks in advance, my apologies if this is something simple, this is my first time trying to do something like this.

Edit: I forgot to mention that my CPU utilization peaks slightly before the GPU utilization.  This led me to think that perhaps it was using the CPU to "read" the file, and then performing the upscaling.  I don't know if that is something that would happen, its just my hypothesis.

**Comment by kodxana** [comment]
For now NCNN models are the fastes model that you can use thanks to NCNN optimization process. PyTorch models do not have optimization with makes model slower. It will be fine on anime models but the base one is very heavy and will be very slow.

**Comment by darkcristle** [comment]
My apologies for re-opening, I'm not sure if it's better etiquette to create a new issue or not.  I switched back to vulkan after reading this and am once again having the issue of low GPU usage.  Here is my understanding of what you said plus my own googling.

NCNN is far better optimized and uses the Vulkan API which allows it to interact directly with the GPU instead of using a separate CUDA environment?  This causes it to be faster.

Issue.  Utilization via task manager never crosses the 10-15% threshold, however when I do something like FFMPEG and have it encode using hardware acceleration this number goes up well above that.  

When I go to Performance under task manager I see that the process I have identified as being the model is using "3d" rather than compute or CUDA.  Is this where it is supposed to be?

I also suspected bottlenecking may be a cause but CPU never rises above 7% so nothing is being outsourced that I can see.

I am running on an intel i9 9900k and a 1080TI FE.  Is there anything I can do to get the utilization up.  
P.S.  I tried messing with -j argument and there was no increase in utilization.

Thanks in advance.
